test_runner/batch_others/test_proxy.py

```diff
@@ -22,6 +29,154 @@ def test_password_hack(static_proxy):
         static_proxy.safe_psql('select 1', sslsni=0, user=user, password=magic)
 
 
+async def get_session_id_from_welcome_message(local_link_proxy, proc):
+    # Read line by line output of psql.
+    # We expect to see result in first 3 lines. [this is flexible, change if need be]
+    # Expected output:
+    #   A notice that contains a welcome to neon message that contains a http url in the form of
+    #   "http://dummy-uri/<psql_session_id>"
+    #   (here dummy-uri was passed from the neon_fixture that created the link_proxy
+    #   The url has to be on its own line.
+    #   Example:
+    #   "
+    #   NOTICE:  Welcome to Neon!
+    #   Authenticate by visiting:
+    #      http://dummy-uri/<psql_session_id>
+    #   "
+
+    psql_session_id = None
+
+    max_num_lines_of_welcome_message = 15
+    attempt = 0
```

Comment:
Misleading variable name. This makes it seem like we're retrying the same thing until it works. But actually we're iterating over the output lines.

Reply:
done